controllers/components/modelcontroller/modelcontroller_actions.go

```diff
@@ -41,6 +41,19 @@ func initialize(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
 		ContextDir: ComponentName,
 		SourcePath: "base",
 	})
+
+	nimState := strings.ToLower(string(operatorv1.Removed))
+	if mc.Spec.Kserve.ManagementState == operatorv1.Managed && mc.Spec.Kserve.NIM.ManagementState == operatorv1.Managed {
+		nimState = strings.ToLower(string(operatorv1.Managed))
+	}
```

Comment:
Can this be be simplified with something like:

```go
nimState := operatorv1.Removed
if c.Spec.Kserve.ManagementState == operatorv1.Managed {
    nimState = mc.Spec.Kserve.NIM.ManagementState
}
```

Reply:
udpated